components/admin-panel/sections/NotificationsSettings/ActivitySwitch.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage } from 'react-intl';
+
+import { ActivityClasses, ActivityTypes } from '../../../../lib/constants/activities';
+import { API_V2_CONTEXT, gqlV2 } from '../../../../lib/graphql/helpers';
+
+import StyledSwitch from '../../../StyledSwitch';
+import { TOAST_TYPE, useToasts } from '../../../ToastProvider';
+
+import { accountActivitySubscriptionsFragment } from './fragments';
+
+const refetchEmailNotificationQuery = gqlV2/* GraphQL */ `
+  query NotificationsSettingsRefetchQuery($id: String!) {
+    account(id: $id) {
+      id
+      ...AccountActivitySubscriptionsFields
+    }
+  }
+  ${accountActivitySubscriptionsFragment}
+`;
+
+const toggleEmailNotificationMutation = gqlV2/* GraphQL */ `
+  mutation ToggleEmailNotification($type: ActivityAndClassesType!, $account: AccountReferenceInput) {
+    toggleEmailNotification(type: $type, account: $account) {
+      id
+    }
+  }
+`;
+
+const ActivitySwitch = ({ account, activityType }) => {
+  const { addToast } = useToasts();
+  const existingSetting = account.activitySubscriptions?.find(
+    notification =>
+      ActivityClasses[activityType] === notification.type || notification.type === ActivityTypes.ACTIVITY_ALL,
+  );
+  const isIndeterminate =
+    activityType === 'ACTIVITY_ALL' &&
+    account.activitySubscriptions?.some(notification => notification.type !== ActivityTypes.ACTIVITY_ALL);
+  const subscribed = existingSetting ? existingSetting.active : isIndeterminate ? false : true;
+  const isOverridedByAll = activityType !== 'ACTIVITY_ALL' && existingSetting?.type === ActivityTypes.ACTIVITY_ALL;
+
+  const [toggleEmailNotification, { loading }] = useMutation(toggleEmailNotificationMutation, {
+    context: API_V2_CONTEXT,
+    refetchQueries: [{ query: refetchEmailNotificationQuery, variables: { id: account.id }, context: API_V2_CONTEXT }],
```

Comment:
Sounds like we don't necessarily need an additional refetch, if the mutation returns an `ActivitySubscriptions` then all we have to do is:

```gql
toggleEmailNotification(...) {
  account {
    ...AccountActivitySubscriptionsFields
  }
}
```

Reply:
The problem here is that sometimes we're deleting the ActivitySubscription, meaning that the returned entity would be null and there would be no way to fetch an account from.

Reply:
Good point. Then returning an `Account` from the mutation might be interesting, but no blocker for me on that one